packages/experiments/src/components/Microfeedback/Microfeedback.view.tsx

```diff
@@ -0,0 +1,178 @@
+
+    return (
+      <div>
+        <Stack horizontal styles={microfeedbackStyles}>
+          <div ref={this.likeRef}>
+            <IconButton menuIconProps={{ iconName: likeIcon }} title={this.props.thumbsUpTitle} onClick={this._vote.bind(this, 1)} />
+          </div>
+          <div ref={this.dislikeRef}>
+            <IconButton menuIconProps={{ iconName: dislikeIcon }} title={this.props.thumbsDownTitle} onClick={this._vote.bind(this, -1)} />
+          </div>
+        </Stack>
+        {this.props.ThumbsUpQuestion ? (
+          <Callout
+            hidden={hideThumbsUpCallout}
+            role="alertdialog"
+            gapSpace={0}
+            target={this.likeRef.current}
+            setInitialFocus={true}
+            onDismiss={onCalloutDismiss}
+          >
+            <div>
+              <FocusZone direction={FocusZoneDirection.vertical}>
+                <p className={styles.title}>{this.props.ThumbsUpQuestion.question}</p>
+                <List items={this.props.ThumbsUpQuestion.options} onRenderCell={this._onRenderCalloutItem} />
+              </FocusZone>
+            </div>
+          </Callout>
+        ) : (
+          <div />
+        )}
+        {this.props.ThumbsDownQuestion ? (
+          <Callout
+            hidden={hideThumbsDownCallout}
+            role="alertdialog"
+            gapSpace={0}
+            target={this.dislikeRef.current}
+            setInitialFocus={true}
+            onDismiss={onCalloutDismiss}
+          >
+            <div>
+              <FocusZone direction={FocusZoneDirection.vertical}>
+                <p className={styles.title}>{this.props.ThumbsDownQuestion.question}</p>
+                <List items={this.props.ThumbsDownQuestion.options} onRenderCell={this._onRenderCalloutItem} />
+              </FocusZone>
+            </div>
+          </Callout>
+        ) : (
+          <div />
+        )}
+      </div>
+    );
+  }
+
+  private _onCalloutDismiss(): void {
+    this.setState({ isFollowupVisible: false });
+  }
+
+  private _onRenderCalloutItem = (item: string, index: number | undefined): JSX.Element => {
+    return (
+      <DefaultButton
+        data-is-focusable={true}
+        styles={microfeedbackItemStyles}
+        text={`${item}`}
+        onClick={this._listOptions.bind(this, index)}
+      />
+    );
+  };
+
+  private _listOptions(option: any): void {
+    this._onCalloutDismiss();
+    if (this.props.sendFeedback) {
+      this.props.sendFeedback(option);
+    }
+  }
+
+  private _vote(vote: number): void {
+    // If the vote that is already selected is picked, then toggle off
+    const updatedVote: number = this.state.vote === vote ? 0 : vote;
+    this.setState({ isFollowupVisible: true, vote: updatedVote });
+    if (updatedVote !== 0 && this.props.sendFeedback) {
+      this.props.sendFeedback(vote);
+    }
+  }
+}
+
+export const MicrofeedbackView: IMicrofeedbackComponent['view'] = props => {
```

Comment:
The intention of state and view components here is that view is just a function, not a class component with state. This is kind of a wrapper workaround going against intent. However, we do want to rework our state/view this approach for hooks, so for now maybe the best thing to do is use `styled` instead of `createComponent`. This component can be converted to use slots later in a backwards compatible way.

This is probably the trickiest item to resolve in this PR. It also depends on what potential slots would be, and I'd like to take a look at how the component works with you. I'm willing to meet with you to work this out.

Reply:
Switched to styled